eng/pipelines/coreclr/templates/build-job.yml

```diff
@@ -33,10 +33,10 @@ jobs:
     condition: ${{ parameters.condition }}
 
     # Compute job name from template parameters
-    ${{ if eq(parameters.useGCC, true) }}:
-      name: ${{ format('coreclr_gcc_product_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
+    ${{ if eq(parameters.compilerName, 'gcc') }}:
+      name: ${{ format('coreclr_{0}_product_build_{1}{1}_{3}_{4}', parameters.compilerName, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
       displayName: ${{ format('CoreCLR GCC Product Build {0}{1} {2} {3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
-    ${{ if eq(parameters.useGCC, false) }}:
+    ${{ if eq(parameters.compilerName, 'clang') }}:
       name: ${{ format('coreclr_product_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
       displayName: ${{ format('CoreCLR Product Build {0}{1} {2} {3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
```

Comment:
I tried to generalize these two branches, but changing `coreclr_product_build_` for clang jobs to `coreclr_clang_product_build_` requires changes in four other files, so I have revert it for now. 🙈

Reply:
Yes, let's not change that 😄